src/uk/gov/defra/ffc/Tests.groovy

```diff
@@ -25,56 +26,90 @@ class Tests implements Serializable {
     }
   }
 
-  static def runZapScan(ctx, projectName, buildNumber, tag) {
+  static runServiceAcceptanceTests(ctx, projectName, serviceName, buildNumber, tag, pr, environment) {
+    ctx.gitStatusWrapper(credentialsId: 'github-token', sha: Utils.getCommitSha(ctx), repo: Utils.getRepoName(ctx), gitHubContext: GitHubStatus.RunTests.Context, description: GitHubStatus.RunTests.Description) {
+      String sanitizedTag = Utils.sanitizeTag(tag)
+      String acceptanceTestService = 'test-runner'
+      try {
+        ctx.sh('mkdir -p -m 777 test-output')
+        if (ctx.fileExists('./docker-compose.migrate.yaml')) {
+          ctx.sh("docker-compose -p $projectName-${sanitizedTag}-$buildNumber -f docker-compose.migrate.yaml run database-up")
+        }
+        ctx.withEnv(Provision.getBuildQueueEnvVars(ctx, serviceName, pr)) {
+          ctx.sh("docker-compose -p $projectName-${sanitizedTag}-$buildNumber -f docker-compose.yaml -f docker-compose.acceptance.yaml run $serviceName-$acceptanceTestService")
+        }
+      } finally {
+        ctx.sh("docker-compose -p $projectName-${sanitizedTag}-$buildNumber -f docker-compose.yaml -f docker-compose.acceptance.yaml down -v")
+        if (ctx.fileExists('./docker-compose.migrate.yaml')) {
+          ctx.sh("docker-compose -p $projectName-${sanitizedTag}-$buildNumber -f docker-compose.migrate.yaml down -v")
+          generateHtmlReport(ctx, projectName)
+        }
+      }
+    }
+  }
+
+  private static generateHtmlReport(ctx, projectName) {
```

Comment:
Should be typed to `void`.

For info, `private` is broken in Groovy so using it has no effect.